spec/fakefs_helper.rb

```diff
@@ -0,0 +1,50 @@
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Metalware.
+#
+# Alces Metalware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Metalware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Metalware, please visit:
+# https://github.com/alces-software/metalware
+#==============================================================================
+require 'fakefs/safe'
+
+class FakeFSHelper
+  def initialize(metal_config)
+    @metal_config = metal_config
+    FakeFS.deactivate!
+    FakeFS.clear!
```

Comment:
Requiring `FakeFS` is error prone. For example `require 'fakefs` will immediately start faking the FS. That is why `require 'fakefs/safe'` needs to be used instead. 

Also because their can only be one faked `FS`, I am not sure how it will behave between the tests. Also turning off the faked FS doesn't delete it. If you turn it back on, then it will be the same as before.

So the `clear!` insures that the faked FS is deleted (i.e. only has the root dir `/`) and the `deactivate!` insures that it isn't running. These two actions are independent and can happen in any order.